src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -273,14 +284,22 @@ static void uct_rdmacm_cm_handle_event_disconnected(struct rdma_cm_event *event)
     struct sockaddr    *remote_addr = rdma_get_peer_addr(event->id);
     char               ip_port_str[UCS_SOCKADDR_STRING_LEN];
     char               ep_str[UCT_RDMACM_EP_STRING_LEN];
+    uct_cm_remote_data_t remote_data;
 
     ucs_debug("%s: got disconnect event, status %d peer %s",
               uct_rdmacm_cm_ep_str(cep, ep_str, UCT_RDMACM_EP_STRING_LEN),
               event->status, ucs_sockaddr_str(remote_addr, ip_port_str,
                                              UCS_SOCKADDR_STRING_LEN));
 
     cep->flags |= UCT_RDMACM_CM_EP_GOT_DISCONNECT;
-    cep->super.disconnect_cb(&cep->super.super.super, cep->super.user_data);
+    if (cep->flags & UCT_RDMACM_CM_EP_GOT_CONNECT) {
```

Comment:
seems uct_rdmacm_cm_ep_error_cb already handles this state check internally, no?

Reply:
it handles `UCT_RDMACM_CM_EP_CONNECTED`